### PR TITLE
Split and rename delete draft service

### PR DESCRIPTION
**Pull request by kevindew**
Trello: https://trello.com/c/yqXt0RGc/1417-remove-documents-at-dwps-request

Following up from https://github.com/alphagov/content-publisher/pull/1755 this separates out some of the responsibilities of the DeleteDraftEditionService to be handled by separate services for discarding path reservations and deleting assets.

As part of this I re-considered the naming of the service and felt that DiscardDraftEditionService was more fitting with the nomenclature we use within the application.

Finally I found and resolved a bug where we updating the edition in the event of re-raising an exception which had no effect as the transaction would be aborted.

More info in commits.

### CODE REVIEW

File: spec/services/delete_draft_edition_service_spec.rb [hunk 1]
```diff
@@ -7,202 +7,103 @@
...
-        expect { described_class.call(edition, user) }
-          .to raise_error(GdsApi::BaseError)
+      expect { described_class.call(edition, user) }
+        .to raise_error(GdsApi::BaseError)
 
-        expect(edition.revision_synced?).to be(false)
-      end
+      expect(edition.revision_synced?).to be(false)
     end
+
```
Comment (emmabeynon) [inline spec/services/delete_draft_edition_service_spec.rb]: Spotted an annoying extra line 🙄

File: spec/services/delete_draft_assets_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+RSpec.describe DeleteDraftAssetsService do
+  describe ".call" do
+    it "attempts to delete an editions assets from asset manager" do
```
Comment (emmabeynon) [inline spec/services/delete_draft_assets_service_spec.rb]: "editions" => "edition's"

File: spec/services/discard_draft_edition_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    end
+
+    it "raises an exception if the current edition is live" do
+      edition = build(:edition, live: true)
+
+      expect { described_class.call(edition, user) }
+        .to raise_error("Trying to delete a live edition")
+    end
+
+    it "changes the editions current flag" do
```
Comment (emmabeynon) [inline spec/services/discard_draft_edition_service_spec.rb]: "editions" => "edition's"

File: spec/services/discard_draft_edition_service_spec.rb [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        .to raise_error(GdsApi::HTTPUnprocessableEntity)
+    end
+
+    it "delegates to the DiscardPathReservationsService for the first edition" do
+      edition = build(:edition)
+      expect(DiscardPathReservationsService).to receive(:call).with(edition)
+      described_class.call(edition, user)
+    end
+
+    it "doesn't discard paths for later editions" do
```
Comment (emmabeynon) [inline spec/services/discard_draft_edition_service_spec.rb]: "editions" => "edition's"

File: spec/services/discard_path_reservations_service_spec.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
+RSpec.describe DiscardPathReservationsService do
+  describe ".call" do
+    it "unreserves an editions path reservations" do
```
Comment (emmabeynon) [inline spec/services/discard_path_reservations_service_spec.rb]: "editions" => "edition's"

**Comment by kevindew** [comment]
Argh! why can't I English? Thanks Emma, will fix.

File: spec/services/discard_draft_edition_service_spec.rb (hunk 4 above)
Comment (kevindew) [reply spec/services/discard_draft_edition_service_spec.rb]: is this one not correct actually? as its plural rather than ownership?

Comment (emmabeynon) [reply spec/services/discard_draft_edition_service_spec.rb]: Yes, time for me to be bad at English 😆

**Pull request merged**